src/univariate/discrete/betabinomial.jl

```diff
@@ -82,18 +82,22 @@ end
 
 function pdf(d::BetaBinomial, k::Int)
     n, α, β = d.n, d.α, d.β
-    choose = 1 / ((n+1) * beta(k+1, n-k+1))
+    chooseinv = (n+1) * beta(k+1, n-k+1)
     numerator = beta(k + α, n - k + β)
     denominator = beta(α, β)
-    return choose * (numerator / denominator)
+    return (numerator / denominator) / chooseinv
 end
 
 function pdf(d::BetaBinomial)
     n, α, β = d.n, d.α, d.β
-    k = 0:n
-    binoms = [1 / ((n+1) * beta(i+1, n-i+1)) for i in k]
-    fixed_beta = beta(α, β)
-    return binoms .* @compat(beta.(k + α, n - k + β)) / fixed_beta
+    denominator = beta(α, β)
+    values = Array{Float64, 1}()
+    for k = 0:n
+        chooseinv = (n+1) * beta(k+1, n-k+1)
+        numerator = beta(k + α, n - k + β)
+        push!(values, (numerator / denominator) / chooseinv)
```

Comment:
Rather than growing the array using `push!`, you know how large you want it to end up, so it's better to allocate the array with the desired length ahead of time. Also note that having `Array{Float64,1}` will implicitly `convert` the elements to `Float64`. In the interest of generality, I think it would be better to write this more like how it was written previously.